src/main/java/org/springframework/data/couchbase/repository/query/support/N1qlUtils.java

```diff
@@ -145,7 +146,7 @@ public static Expression createWhereFilterForEntity(Expression baseWhereCriteria
                                                       EntityMetadata<?> entityInformation) {
     //add part that filters on type key
     String typeKey = converter.getTypeKey();
-    String typeValue = entityInformation.getJavaType().getName();
+    String typeValue = getTypeValue(entityInformation);
```

Comment:
I didn't understand when you said : "unless the documents are correctly serialized with the correct type value"?
Actually, the method I have created covers the 2 cases : 
* When @TypeAlias was used when serializing an object. It would be then logical to be also used when deserializing.
* Else, we'll use the previous mapping, ie, the QFN of the class.

For the other String, Spatial and view queries, I'll check.